libkbfs/block_retrieval_worker_test.go

```diff
@@ -251,3 +258,46 @@ func TestBlockRetrievalWorkerShutdown(t *testing.T) {
 	w.Shutdown()
 	require.True(t, shutdown)
 }
+
+func TestBlockRetrievalWorkerMultipleBlockTypes(t *testing.T) {
+	t.Log("Test that we can retrieve the same block into different block types.")
+	codec := kbfscodec.NewMsgpack()
+	q := newBlockRetrievalQueue(1, codec)
+	require.NotNil(t, q)
+	defer q.Shutdown()
+
+	bg := newFakeBlockGetter()
+	w1 := newBlockRetrievalWorker(bg, q)
+	require.NotNil(t, w1)
+	defer w1.Shutdown()
+
+	t.Log("Setup source blocks")
+	ptr1 := makeFakeBlockPointer(t)
+	block1 := makeFakeFileBlock(t)
+	ch1 := bg.setBlockToReturn(ptr1, block1)
+	testCommonBlock := &CommonBlock{}
+	bytes, err := codec.Encode(block1)
```

Comment:
Again you can probably use `kbfscodec.Update` here instead of two separate calls to Encode/Decode.